metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/loader/xml/SraAnalysisXmlParser.java

```diff
@@ -28,9 +28,12 @@
 
     private static final Logger LOGGER = Logger.getLogger(SraAnalysisXmlParser.class.getName());
 
+    private final String XML_SET_TAGS = "(<ANALYSIS_SET>|</ANALYSIS_SET>)";
+
     @Override
     public AnalysisType parseXml(String xmlString, String accession) throws XmlException {
-        xmlString = removeRootTagsFromXmlString(xmlString);
+        xmlString = removeRootTagsFromXmlString(xmlString); // For API calls
+        xmlString = removeSetTagsFromXmlString(xmlString); // For database queries
```

Comment:
I wonder whether we should remove the root/set tags in the `SraRetrieverByAccession` concrete classes, so that the XML strings returned from `getXml` methods are well-formed XML strings for the corresponding object types.

Reply:
I wanted to keep the responsibilities of each class as delimited as possible, in the sense that one would be only for querying and another for parsing XML (including all its considerations). And because it is hard to implement a 100%-guaranteed generic `removeSetTagsFromXmlString` method (see other comment) I'm not sure it will be highly beneficial to move to `SraRetrieverByAccession`.